src/app/models/Recipe.ts

```diff
@@ -0,0 +1,21 @@
+export class Recipe {
+    key: number;
+    name: string;
+    description: string;
+    imageUrl: string;
+    featured: boolean;
+    liked: boolean;
+    saved: boolean;
+    tags: Array<string>;
+
+    constructor(obj: Recipe) {
+        this.key = obj.key;
```

Comment:
what if by mistake you send null instead of obj? make sure first the obj is present